app/src/main/java/com/example/tarok/views/PlayButtonsView.java

```diff
@@ -26,15 +60,234 @@ public PlayButtonsView(MainActivity mainActivity) {
         this.playSoloTwo = mainActivity.findViewById(R.id.soloTwo);
         this.playSoloOne = mainActivity.findViewById(R.id.soloOne);
 
-        setupButtonOnClick();
+        this.skipButton = mainActivity.findViewById(R.id.skipButton);
+
+        this.playThreeLabel = mainActivity.findViewById(R.id.playThreeLabel);
+        this.playTwoLabel = mainActivity.findViewById(R.id.playTwoLabel);
+        this.playOneLabel = mainActivity.findViewById(R.id.playOneLabel);
+        this.playSoloThreeLabel = mainActivity.findViewById(R.id.playSoloThreeLabel);
+        this.playSoloTwoLabel = mainActivity.findViewById(R.id.playSoloTwoLabel);
+        this.playSoloOneLabel = mainActivity.findViewById(R.id.playSoloOneLabel);
+
+        this.bidInformerLabel = mainActivity.findViewById(R.id.bidInformerLabel);
+
+        buttonList = new ArrayList<>();
+
+        buttonList.add(playThree);
+        buttonList.add(playTwo);
+        buttonList.add(playOne);
+        buttonList.add(playSoloThree);
+        buttonList.add(playSoloTwo);
+        buttonList.add(playSoloOne);
+
+        textViewList = new ArrayList<>();
+
+        textViewList.add(playThreeLabel);
+        textViewList.add(playTwoLabel);
+        textViewList.add(playOneLabel);
+        textViewList.add(playSoloThreeLabel);
+        textViewList.add(playSoloTwoLabel);
+        textViewList.add(playSoloOneLabel);
+
+        this.decks = decks;
+
+        setUpBiddingProcess();
+    }
+
+    /**
+     * Allows the human player to make a bid, and initiates the bot bidding process
+     */
+    private void setUpBiddingProcess(){
+        skipButton.setOnClickListener(view -> {
+            if(currentLowestBid == 0){
+                bidInformerLabel.setText("PLEASE MAKE THE FIRST BID");
+            } else {
+                skips++;
+                bidInformerLabel.setText("PLAYER 1 SKIPS");
+                if (skips == 3) {
+                    sendPlayModeToMain(2);
+                } else {
+                    makeBotBids(0);
+                }
+            }
+        });
+        playThree.setOnClickListener(view -> {
+            currentLowestBid = 0;
+            playThreeLabel.setText("PLAYER 1");
+            enableAllButtons(false);
+            displayLatestBid();
+            skips = 0;
+            makeBotBids(0);
```

Comment:
We should replace all of these with a ClickListener class which takes currentLowestBid as a parameter.

Reply:
Definitely, the code duplication is horrible here. However, makeBotsBid(int botId) handles some UI stuff from the PlayButtonsView (colours the buttons), so we'll have to refactor that as well to make this possible